### ISSUE TITLE
DragonFlight 10.0 Fixes

### ISSUE THREAD

**Issue by Ryan-Rogue** [varren/ViragDevTool#12]
ViragDevTool.XML:
combine lines 316 and 317 into this:
                
                 self:SetResizeBounds(635, 200, UIParent:GetWidth(), UIParent:GetHeight());

ViragDevTool.lua:
Combine lines 514 and 515 into this:

                  local minX, minY, maxX, maxY = parentFrame:GetResizeBounds()

Should clear the Lua errors and give you a functioning UI again.

**Comment by brittyazel** [comment]
You can you use my fork if you like. It's fully updated for Dragonflight among many other things.

https://github.com/brittyazel/DevTool

**Comment by mrbuds** [comment]
> You can you use my fork if you like. It's fully updated for Dragonflight among many other things.
> 
> https://github.com/brittyazel/ViragDevTool

I made my own fixes for 10.0 and was going to ask varren permission to publish my changes, then i saw you made your own fork.

Do you intend to publish on wago/curseforge?
If you want to publish then i could help (if you need) with using bigwigs packager to get updated libs from external source to build a usable archive.
Ho and by the way, would you activate the issue panel on your own github?

**Comment by brittyazel** [comment]
I'm open to the idea of publishing it with Varen's blessing. I believe his license is all-rights-reserved, so I'd like a bit of clarity there.

**Comment by brittyazel** [comment]
Issue panel enabled! And yeah, if I were to publish it I would totally use the bigwigs packager like I do on my other addons :-) Any help is appreciated though!

My fork is more than just a few fixes. I completely ported to the Ace3 framework and retooled many things internally to try to make it easier (at least for me) to make heads or tails of.

**Comment by JoschiZ** [comment]
I would really appreciate your fork being available for auto updates somehow.
I hope Varren gives you his blessing. Sadly it seems like he has gone completely silent.

**Comment by N6REJ** [comment]
when you scroll thru the databases periodically bug grabber will start going nuts and will returns thousands of errorrs.

**Comment by brittyazel** [comment]
@N6REJ on my version or varrens? Please try my version and tell me if it works better.

**Comment by N6REJ** [comment]
on yours.. I did a fresh clone this morning.
Hit G and start dragging down about 1/3rd way and then it will trigger.

**Comment by N6REJ** [comment]
https://github.com/N6REJ/Bears_Audio_Switcher  This is the addon I was searching the db for as I'm having an issue where it will randomly forget the options for spkr 2 that were already selected.  It's when I got here it went crazy.

**Comment by N6REJ** [comment]
ping me here https://discord.gg/cnsgGN7t if you need/want my rapid attention

**Comment by brittyazel** [comment]
Please report the bug on the issue tracker on my repository, and please post the Lua error. I will take a look when I can.

**Comment by N6REJ** [comment]
ok, I'll do it late tonight.

**Comment by brittyazel** [comment]
Nvm I think I found and fixed it. Please report back if the issue is solved. In the future, please report all new issues as their own ticket, rather than as a comment on another ticket. And, please report issues pertaining to my version on my own repository. Cheers.

**Comment by srhinos** [comment]
a bit late to this but I've also been maintaining a fork over at [here](https://github.com/curseforge-mirror/ViragDevTool), mainly some small fixes layered on top of major ones by @brittyazel and a few others I found in this repo's [network tab](https://github.com/varren/ViragDevTool/network).

I set up the bw packager to auto release updates there but would also be more than happy to see those changes merged into @brittyazel's branch as they have a better grasp on maintaining this addon than I do long term.

**Comment by brittyazel** [comment]
Feel free to send me some PRs and I'll get them merged asap 🙂

**Comment by brittyazel** [comment]
I finally got around to working on my forked version, and I'm about ready to release it under the new name of "DevTool". I have gone though and rewritten and optimized huge swaths of the core code, and in my testing it's now about 90% more efficient with regards to CPU and memory usage when navigating large tables (such as _G). I've also cleaned up and polished a lot of the core features, fixed a ton of bugs, and have added some usability features.

Stay tuned for it to show up on CurseForge and WoWUp

https://github.com/brittyazel/DevTool

**Comment by brittyazel** [comment]
Aight my version, named [DevTool](https://github.com/brittyazel/DevTool), has it's first release out v1.0.1, and is up on [Curseforge ](https://legacy.curseforge.com/wow/addons/devtool/files)and [Wago](https://addons.wago.io/addons/devtool). (It might take a minute to get approved on Curseforge, if you don't see it).

**Comment by varren** [comment]
> Aight my version, named [DevTool](https://github.com/brittyazel/DevTool), has it's first release out v1.0.1, and is up on [Curseforge ](https://legacy.curseforge.com/wow/addons/devtool/files)and [Wago](https://addons.wago.io/addons/devtool). (It might take a minute to get approved on Curseforge, if you don't see it).

Good job @brittyazel i was kinda out of github for a while coz of Russian-Ukraine war with all the events that followed. I'll add a note that my version is deprecated and put a link to your github in the readme.

**Comment by brittyazel** [comment]
Thanks so much! I appreciate you doing that 🙂. I hope all is well with you and your family.

**Comment by varren** [comment]
@brittyazel Yeah all friends and family on both sides are fine, considering the circumstances.  Feels surreal. Most thoughts are about the conflict and the future and that's why it's hard to concentrate on side activities. 

Gl with maintaining the addon.
